### PR TITLE
update xctl

### PR DESCRIPTION
**Pull request by Frankonly**

### CODE REVIEW

File: ioctl/cmd/root.go [hunk 1]
```diff
@@ -24,50 +22,80 @@ import (
...
+		config.TranslateInLang(flagOutputFormatUsages, config.UILanguage))
+
+	return rootCmd
 }
 
-// Execute adds all child commands to the root command and sets flags appropriately.
-func Execute() {
-	if err := RootCmd.Execute(); err != nil {
-		os.Exit(1)
+// NewXctl returns ioctl root cmd
```
Comment (dustinxie) [inline ioctl/cmd/root.go]: return xctl root cmd

File: tools/xctl/readme/docgen.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+	path := filepath.Join(xctlPath, "readme")
+	err = GenMarkdownTreeCustom(rootCmd, path, filePrepender, linkHandler)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// GenMarkdownTreeCustom is the the same as GenMarkdownTree, but
+// with custom filePrepender and linkHandler.
+func GenMarkdownTreeCustom(c *cobra.Command, dir string, filePrepender func(string) string,
```
Comment (dustinxie) [inline tools/xctl/readme/docgen.go]: is it possible to reuse code? it looks same as docgen.go under /ioctl

File: ioctl/cmd/account/accountcreate.go [hunk 3]
```diff
@@ -63,8 +63,10 @@ type generatedAccount struct {
 func init() {
 	accountCreateCmd.Flags().UintVarP(&numAccounts, "num", "n", 1,
 		config.TranslateInLang(flagNumUsages, config.UILanguage))
+
 	accountCreateCmd.Flags().BoolVar(&CryptoSm2, "sm2", false,
 		config.TranslateInLang(flagSm2Usage, config.UILanguage))
+	accountCreateCmd.Flags().MarkHidden("sm2")
```
Comment (CoderZhi) [inline ioctl/cmd/account/accountcreate.go]: I thought we agree to delete these

File: ioctl/cmd/account/accountcreateadd.go [hunk 4]
```diff
@@ -46,6 +46,8 @@ var accountCreateAddCmd = &cobra.Command{
 func init() {
 	accountCreateAddCmd.Flags().BoolVar(&CryptoSm2, "sm2", false,
 		config.TranslateInLang(flagSm2Usage, config.UILanguage))
+	accountCreateAddCmd.Flags().MarkHidden("sm2")
+
 }
```
Comment (CoderZhi) [inline ioctl/cmd/account/accountcreateadd.go]: same here, CryptoSm2 won't be changed once command is compiled

Comment (Frankonly) [reply ioctl/cmd/account/accountcreateadd.go]: ok

**Pull request merged**